src/tail/tunnel.rs

```diff
@@ -34,28 +32,26 @@ impl Tunnel {
         let command_name = format!("{:?}", command);
 
         let child = command
-            .kill_on_drop(true)
             .spawn()
             .expect(&format!("{} failed to spawn", command_name));
 
-        Ok(Tunnel {
-            child,
-            command_name,
-        })
+        Ok(Tunnel { child })
+    }
+
+    pub async fn run(self, rx: Receiver<()>) -> Result<(), failure::Error> {
+        rx.await?;
+        self.shutdown().await
     }
 
-    pub async fn run(self) -> Result<(), failure::Error> {
-        let status = self.child.await?;
+    pub async fn shutdown(mut self) -> Result<(), failure::Error> {
+        // eprintln!("killing cloudflared");
+        if let Err(e) = self.child.kill() {
+            failure::bail!("failed to kill cloudflared: {}", e)
```

Comment:
this shouldn't happen but if it does we want to know. cloudflared will eventually exit, but we could also instruct the user to just `kill <child.id>`